internal/upload/upload_test.go

```
@@ -0,0 +1,724 @@
+}
+
+func TestCommitSkipsCollisionsUnlessOverwriteRequested(t *testing.T) {
+	for _, overwrite := range []bool{false, true} {
+		name := "skip"
+		if overwrite {
+			name = "overwrite"
+		}
+		t.Run(name, func(t *testing.T) {
+			m, root := newTestManager(t)
+			dest := filepath.Join(root, "Artist", "01.flac")
+			if err := os.MkdirAll(filepath.Dir(dest), 0o755); err != nil {
```

golangci-lint (2.12.2)

[medium] G301: Expect directory permissions to be 0750 or less

(gosec)